content/ui-patterns/degraded-experiences.mdx

```diff
@@ -0,0 +1,518 @@
+
+An inert button is the accessible option because buttons may not be disabled if they respond to user input. For example, if a button shows a tooltip when hovered or focused, or a dialog when clicked.
+
+</div>
+
+<img
+  width="456"
+  alt="A default button and an inert button"
+  src="https://github.com/primer/design/assets/2313998/fd1a7c99-b8c3-43fd-9f74-ae509e9efa31"
+/>
+
+</Box>
+
+There are two recommended ways an inert button should respond to user input:
+
+1. **Show a dialog on click:** When a user tries to activate an inert the button, open a [dialog](/components/dialog) that explains why the action cannot be completed and give them a path forward if possible. It is required to provide some kind of feedback when a user clicks a button.
+2. **Show a tooltip on hover or focus:** Before a user tries to activate a non-functional control, a [tooltip](/components/tooltip) with additional context may be displayed on hover or focus. It is _not_ required to respond to hover and focus.
+
+<img
+  width="960"
+  alt="A video of a 'broken' button that shows a tooltip and opens an error dialog when activated"
+  src="https://github.com/primer/design/assets/2313998/7786d3f9-c70c-4109-a49d-549231ccf79d"
+/>
+
+#### Disabling a non-functional button
+
+{/* <!-- TODO: Come up with a visual example where it's safe to disable the button. --> */}
+
+<img
+  width="960"
+  alt="Example of a button that can be disabled because it has sufficient context explaining why it is disabled"
+  src="https://github.com/primer/design/assets/2313998/e8ec073d-b0c9-4f41-aa6c-dae29aaf7fca"
+/>
+
+If a button is non-functional and there is already enough context for the user to understand why it's disabled, there's no need to respond to user input, so the button can safely be disabled.
+
+For more info, see the guidance on [disabling links and buttons](https://accessibility-playbook.github.com/link-and-button-guidance#disabling-links-and-buttons) from GitHub's Accessibility Playbook (only available to GitHub staff).
+
+### Handling action menus with non-functional items
+
+<img
+  width="960"
+  alt="Three examples: 1. An action menu that works just fine; 2. An action menu with some items disabled; 3. An action menu with all items disabled"
+  src="https://github.com/primer/design/assets/2313998/92ab5cea-74a8-43f0-834c-3ef07ac65bfe"
+/>
+
+Show the menu items, but disable them and show a message explaining why they're disabled.
+
+If an action menu contains menu items that are very closely related (for example: the action menu for adding a file to a repo) and we can detect that none of the menu items are available, then indicate a problem on the button that triggers the action menu. Use an icon in the button to indicate that something is wrong, and (optionally) show a tooltip on hover or focus.
+
```

Comment:
It's unclear if we'll have any use cases for the following components, so I'm leaving them out until we do: Tabs, Form controls, Links, Segmented controls, Toggle switches

Reply:
Yes, this makes sense. I'd err on the side of adding guidance when/as questions get asked.